openwisp_radius/base/models.py

```diff
@@ -515,6 +515,24 @@ class Meta:
     def __str__(self):
         return self.unique_id
 
+    @classmethod
+    def close_stale_sessions(cls, days):
+        older_than = timezone.now() - timedelta(days=days)
+        # If the "update_time" is recent, then the session is not closed
+        # even when the "start_time" is older than the specified time.
+        # The "start_time" of a session is only checked when the
+        # "update_time" is not set.
+        sessions = cls.objects.filter(
+            Q(update_time__lt=older_than)
+            | (Q(update_time=None) & Q(start_time__lt=older_than))
+        )
+        for session in sessions:
+            # calculate seconds in between two dates
+            session.session_time = (now() - session.start_time).total_seconds()
+            session.stop_time = now()
+            session.update_time = session.stop_time
```

Comment:
one minor request which I had forgot to ask, if the termination cause field is empty, can you please fill it with `Session Timeout`?